app/models/financialDetails/DocumentDetail.scala

```diff
@@ -158,6 +163,22 @@ case class DocumentDetail(taxYear: Int,
     case _ => false
   }
 
+  def isBalancingCharge(codedOutEnabled: Boolean = false): Boolean = getChargeTypeKey(codedOutEnabled) == "balancingCharge.text"
+
+  def isBalancingChargeZero(codedOutEnabled: Boolean = false): Boolean = {
+    (isBalancingCharge(codedOutEnabled), this.originalAmount) match {
+      case (true, Some(value)) => if (value == BigDecimal(0.0)) true else false
```

Comment:
Could simplify: no need for if/else, could evaluate value == BigDecimal(0.0) directly

Reply:
right!! :)